### DISCUSSION TITLE
Super impressed

### DISCUSSION THREAD

**Discussion by MengLinMaker** [spessasus/SpessaSynth#48] [General]
This is a very impressive project, having worked on related stuff.

I notice SoundFont3 by MuseScore does not comply with the RIFF specification.
 But It is what we have to work with now.
 
On a side note, the package bundle size is [375kb gzip](https://bundlejs.com/?q=spessasynth_lib%403.20.22&treeshake=%5B%7B+Synthetizer+%7D%5D). I believe it's possible to shave another ~ 60kb by replacing [stbvorbis_sync.min.js](https://github.com/spessasus/SpessaSynth/blob/master/src/spessasynth_lib/externals/stbvorbis_sync/stbvorbis_sync.min.js) with [BaseAudioContext.decodeAudioData() web api](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/decodeAudioData). Possibly more performant too.

The sound isn't as rich as libfluidsynth, but this is the best javascript synth implementation I've seen so far.

Note: I'm looking to slim down the synth portion of aa web app I created.

**Comment by spessasus** [comment]
Hi, thanks for the kind words! I'm glad you like SpessaSynth.

As for the vorbis decoder, the answer is sadly a no because of two reasons:
1. decodeAudioData is async, which won't work with the synth as it is coded synchronously because of the rendering to file support.
2. The entire synthesizer and MIDI sequencer both run in the audio worklet thread. It has neither access to its own context, not even AudioContext constructor itself.

As for the richness, I'm assuming you're taking about the quality of built-in reverb and chorus, because otherwise it sounds identical (according to my testing at least), even the filter acts like fluid's.

Yes, they are not very good, and I'm open to suggestions on improving these! If you find of create a good sounding configuration (or implementation) for them, let me know and I'll add it. (Or open a PR :-)
